python/cudaq/kernel/ast_bridge.py

```diff
@@ -718,14 +718,26 @@ def visit_Call(self, node):
                                        TypeAttr.get(iTy),
                                        seqSize=totalSize).result
 
+                # Logic here is as follows:
+                # We are building an array like this
                 # array = [start, start +- step, start +- 2*step, start +- 3*step, ...]
+                # So we need to know the start and step (already have them),
+                # but we also need to keep track of a counter
+                counter = cc.AllocaOp(cc.PointerType.get(self.ctx, iTy),
+                                      TypeAttr.get(iTy)).result
+                cc.StoreOp(zero, counter)
+
                 def bodyBuilder(iterVar):
-                    tmp = arith.MulIOp(iterVar, stepVal).result
+                    loadedCounter = cc.LoadOp(counter).result
+                    tmp = arith.MulIOp(loadedCounter, stepVal).result
                     arrElementVal = arith.AddIOp(startVal, tmp).result
                     eleAddr = cc.ComputePtrOp(
-                        cc.PointerType.get(self.ctx, iTy), iterable, [iterVar],
+                        cc.PointerType.get(self.ctx, iTy), iterable,
+                        [loadedCounter],
                         DenseI32ArrayAttr.get([-2147483648], context=self.ctx))
                     cc.StoreOp(arrElementVal, eleAddr)
+                    incrementedCounter = arith.AddIOp(loadedCounter, one).result
```

Comment:
If this is decrementing is the step "normalized" to be non-positive?

Reply:
They way I have it implemented is that I keep track of an incrementing counter, so this Value here should be always incrementing by one. But the step Value could be a negative or positive integer. So 
```python
tmp = arith.MulIOp(loadedCounter, stepVal).result
arrElementVal = arith.AddIOp(startVal, tmp).result
```
will increment or decrement based on whatever the step value is. I'm trying to create this 
```python
array = [start, start +- step, start +- 2*step, start +- 3*step, ...]
#- or -
array = [start +- counter * step ...] 
```
so step is the only thing that determines that `+-`

Reply:
So, that sounds like a yes.

I find the `+-` notation confusing since, as you say, the counter is monotonically increasing. Maybe a notation like

```
array := [start + (step * 0), start + (step * 1), start + (step * 2) ... ] where step is non-zero
```
would be clearer?

(One can also transform these into an exact count, but beware there are incorrect ways of doing it all over the internet.)